src/librustc_lint/unused.rs

```diff
@@ -23,7 +26,7 @@ use log::debug;
 
 declare_lint! {
     pub UNUSED_MUST_USE,
-    Warn,
+    Deny,
```

Comment:
Should we make this deny by default in a separate PR? I'm a bit weary of simultaneously making this broader and more constraining (unless this is just for crater's sake, in that case carry right along).

Reply:
That's just for crater purposes.